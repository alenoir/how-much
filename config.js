import React from 'react';
import ReactNative from 'react-native';
const Immutable = require('immutable');

const window = global || window;

window.React = React;
window.ReactNative = ReactNative;
window.Immutable = Immutable;
