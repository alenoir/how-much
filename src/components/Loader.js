const {
  StyleSheet,
  View,
  Text,
} = ReactNative;

const {
  Component,
} = React;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#2DB0CD',
  },
  loaderContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  loader: {
    fontFamily: 'Varela Round',
    fontSize: 30,
    color: 'white',
  },
});

class Loader extends Component {

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.loaderContainer}>
          <Text style={styles.loader}>Loading...</Text>
        </View>
      </View>
    );
  }

}

module.exports = Loader;
