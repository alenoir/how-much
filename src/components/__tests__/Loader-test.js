import renderer from 'react-test-renderer'; // eslint-disable-line

import '../../../config.js';
import Loader from '../Loader';

describe('Loader', () => {
  it('Render', () => {
    const tree = renderer.create(
      <Loader />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
