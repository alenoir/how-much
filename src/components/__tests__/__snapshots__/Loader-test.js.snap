exports[`Loader Render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2DB0CD",
      "flex": 1
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center"
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "Varela Round",
          "fontSize": 30
        }
      }>
      Loading...
    </Text>
  </View>
</View>
`;
